src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/RazorFormattingService.cs

```diff
@@ -240,4 +242,44 @@ private static void UnwrapCSharpSnippets(TextEdit[] snippetEdits)
             snippetEdit.NewText = unwrappedText;
         }
     }
+
+    // This method counts the occurrences of CRLF and LF line endings in the original text. If LF line endings are more
+    // prevalent, it removes any standalone CR characters from the text edits to ensure consistency with the LF style.
+    // This can be removed once we figure out how to get proper line ending information from the client.
+    private TextEdit[] NormalizeLineEndings(SourceText originalText, TextEdit[] minimalEdits)
+    {
+        var (crlfCount, lfCount) = CountLineEndings(originalText);
+        if (lfCount > crlfCount)
```

Comment:
Out of interest I had a look for how the VS editor does this, and it seems like the impl is here: https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=/src/Editor/Text/Def/TextData/Document/NewlineState.cs&version=GBmain&line=114&lineEnd=115&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

It's basically just assuming the before state is well formed, so a single CRLF means the document in CRLF.

Not suggesting this needs to change though, just thought I'd mention it out of curiosity.

Reply:
Thank you for looking into this! Very interesting implementation indeed but I guess it makes sense.